### DISCUSSION TITLE
[Feature] local backup

### DISCUSSION THREAD

**Discussion by gl-dev** [TechbeeAT/jtxBoard#922] [Ideas]
Local auto backup of data with optional encryption would be great for offline usage with file name supporting YYYY-MM-DD pre-/suffix.

**Comment by patrickunterwegs** [comment]
Hi @gl-dev , let's think about it, but I'm really wondering what's the best way to do something like this respectively Hof far such a functionality would need to go:

The base requirement is a regular export of collections in a predefined folder. Additionally: 
- would there be the need to configure the schedule for the backup (how often, at what time, only when charging,...)? How fine-grained?
- about encryption: would the user choose a key or would the app just encrypt the file?
- is there the need to choose which collections to backup?
- if yes, is there the need to define multiple backup jobs with different settings?

So I'm a bit hesitant with that base requirement, I'm afraid once it's done, much more will be requested ending up on with a request to also import files regularly (and then it gets really complicated and I wouldn't want to do that). 

What's your opinion on this?

**Comment by gl-dev** [comment]
Hello @patrickunterwegs, I just thought about a simple db dump of all the stored user data to a file.
Being able to create a local backup is IMHO a crucial feature to use an app offline.
Encryption and automatic backup is optional, but nevertheless sensible options.
jtxBoard would be an interesting tool to store my data, but for now, just for offline usage - so for me local backup support is definitely a show stopper.

**Comment by patrickunterwegs** [comment]
Hey @gl-dev , 
When it comes to a simple export, you can do this already in the collections overview in the top menu!
It's not a db dump, but an export as an ics file (which is anyway better as it's compliant to the standard!) 😉

**Comment by gl-dev** [comment]
Great, I'll check it out. Thank you!

**Comment by patrickunterwegs** [comment]
I'm moving this topic to "discussions" to keep it open for further discussion

**Comment by boxallar** [comment]
Would it be possible to backup the jtx-board folder via e.g. SyncThing?
Or automate the saving as a .ics file and then syning that with syncthing or similar?
